### DISCUSSION TITLE
Auro REST controller to a service

### DISCUSSION THREAD

**Discussion by meirkr** [canton7/RestEase#218] [General]
Hi
In order to make this amazing library be "full" code first approach, similar to GRPC code first, 
I would like to suggest an extension to this library, which will auto (source) generate service REST controllers to serve the interfaces with all the GET/POST.. definitions on the "code first" interfaces.

suppose we have the interface:
```
public interface ISomeApi
{
    [Get("search")]
    Task<SearchResult> SearchAsync([Query("filter")] string filter1, [Query("filter")] string filter2);
}
```

so it will be very easy to have a service implementing this and maybe mark it with some attribute:
```
[AutoController]
public class SomeApiService : ISomeApi
{
    public async Task<SearchResult> SearchAsync(string filter1, string filter2)
    {
        // ...
        // ...
        // ...
        return new SearchResult(........);
    }
}
```
and the framework somehow will expose a rest controller based on the attributes defined in the interface, hopefully generated code, so swagger could expose it as regular manually written controllers.

@canton7 
What do you think?

**Comment by canton7** [comment]
Sounds great, but, it's an pretty large project which I'm not sure there's a demand for, which I personally have no need for (and interest in), and for which there are workable solutions already (e.g. what Swagger already offers).

If you want to have a stab for your own interest, please feel free: I don't think you'll lose anything by writing a library which depends on RestEase that can do this. I'll happily link to it from the README.

**Comment by ArthurSett** [comment]
Had the same idea: https://github.com/ArthurSett/RestEaseInterface
